### DISCUSSION TITLE
Load dts files per model

### DISCUSSION THREAD

**Discussion by Nuthin2do** [CodinGame/monaco-vscode-api#512] [Q&A] [answered]
Hi,

after i read this thread #498 i wonder, is it possible to load dts files per model and dynamically via parameter with the intellisense implementation (like in the demo)? If it is, do i still need the `crossOriginIsolated` flag since i want to limit intellisense to a model and don't want it project-wide? And can you provide a simple example how to do that or give me some directions on how to do that?

With _"dynamically via parameter"_ i mean that i want to load the dts files in a C# application and send it to the editor api which uses your api

Thanks for your help in advance.

**Comment by CGNonofr** [comment]
Hello,

I'm not sure what you mean by `dts files per model`. dts files are global and shared by all source files in both the monaco worker and the VSCode extension

With the VSCode extension and if crossOriginIsolated is not enabled, I'm not sure your source file will be able to reference to .d.ts next to it :thinking:

What I would do if you can't enable crossOriginIsolated and you only need a single source file is to simply use the monaco worker

**Reply by Nuthin2do** [reply]
What i mean with "dts files per model" is that i would like to load dts files not globally, but per model. Meaning when i have multiple models i would like to have each model with its own set of dts files. Would that be possible? Or would i have to load and then remove the dts files on each change of active model?

**Reply by CGNonofr** [reply]
I don't think it's possible.

What is maybe possible is to not register the type globally by use [reference triple--slash directives](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html#-reference-path-)

**Reply by Nuthin2do** [reply, marked as answer]
Hi, thanks for your answer. I think i found a solution for my objective. I just use setExtraLib of the monaco-lib (-> https://microsoft.github.io/monaco-editor/typedoc/interfaces/languages.typescript.LanguageServiceDefaults.html#setExtraLibs). This fulfills my needs.
